adafruit_pybadger/pybadge.py

```diff
@@ -0,0 +1,123 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_pybadger.pybadge`
+================================================================================
+
+Badge-focused CircuitPython helper library for PyBadge, PyBadge LC and EdgeBadge.
+All three boards are included in this module as there is no difference in the
+CircuitPython builds at this time, and therefore no way to differentiate
+the boards from within CircuitPython.
+
+
+* Author(s): Kattni Rembor
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+* `Adafruit PyBadge <https://www.adafruit.com/product/4200>`_
+* `Adafruit PyBadge LC <https://www.adafruit.com/product/3939>`_
+* `Adafruit EdgeBadge <https://www.adafruit.com/product/4400>`_
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+
+from collections import namedtuple
+import board
+import digitalio
+import analogio
+import audioio
+from gamepadshift import GamePadShift
+import adafruit_lis3dh
+from adafruit_pybadger.pybadger_base import PyBadgerBase
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_PyBadger.git"
+
+Buttons = namedtuple("Buttons", "b a start select right down up left")
+
+class PyBadge(PyBadgerBase):
+    """Class that represents a single PyBadge, PyBadge LC, or EdgeBadge."""
+
+    _audio_out = audioio.AudioOut
+    _neopixel_count = 5
+
+    def __init__(self):
+        super().__init__()
+
+        i2c = None
+
+        if i2c is None:
+            try:
+                i2c = board.I2C()
+            except RuntimeError:
+                self._accelerometer = None
+
+        if i2c is not None:
+            int1 = digitalio.DigitalInOut(board.ACCELEROMETER_INTERRUPT)
+            try:
+                self._accelerometer = adafruit_lis3dh.LIS3DH_I2C(i2c, address=0x19, int1=int1)
+            except ValueError:
+                self._accelerometer = adafruit_lis3dh.LIS3DH_I2C(i2c, int1=int1)
+
+        self._buttons = GamePadShift(digitalio.DigitalInOut(board.BUTTON_CLOCK),
+                                     digitalio.DigitalInOut(board.BUTTON_OUT),
+                                     digitalio.DigitalInOut(board.BUTTON_LATCH))
+
+        self._light_sensor = analogio.AnalogIn(board.A7)
```

Comment:
Is there a way to check if the light sensor isn't present, like for the pybadge LC?

Reply:
PyBadge LC has a light sensor. It's at the top, next to the reset button, with a hole over it to pass the front of the sensor through to the front of the board. From the back, it is white with a green symbol printed on it.

The lack of a light sensor would only be detectable in very specific situations (depending on whether the pin is connected to something else, has pull up or down, etc), and is a complicated process to do so. Knowing that the CLUE doesn't have the same light sensor, I included it in the `unsupported` list within the CLUE module. In the case of PyBadge vs. PyBadge LC, there's no way to tell the difference as they use the same CircuitPython build - there is no separate pin definition for PyBadge LC.

Reply:
D'oh, I meant the accelerometer, not the light sensor. 

If it's possible to auto detect that, then this line from the example could be modified or removed to make the experience on a PyBadge LC a little nicer.

https://github.com/adafruit/Adafruit_CircuitPython_PyBadger/blob/bf25d8b9ddc1c2dee03db6bb486d5fa7bf61b587/examples/pybadger_simpletest.py#L6

Reply:
Try checking what `pybadger.acceleration` returns on the LC. I think it might end up as `None` on the but I don't have one to check with. 

Along the same idea of improving the auto_dim: what if pressing any button would reset the dim timer. That would help all of the boards and allow for at least one type of auto_dim on the LC device.